presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PushPartialAggregationThroughExchange.java

```diff
@@ -126,8 +129,7 @@ public Result apply(AggregationNode aggregationNode, Captures captures, Context
         if (!decomposable ||
                 partialAggregationStrategy == NEVER ||
                 partialAggregationStrategy == AUTOMATIC &&
-                        partialAggregationNotUseful(aggregationNode, exchangeNode, context) &&
-                        aggregationNode.getGroupingKeys().size() == 1) {
```

Comment:
I moved this check to the function partialAggregationNotUseful
The original PR which introduced cost-based reasoning for partial aggs was conservative and only allowed skipping partial aggs for single-column GROUP BY-s because of possible estimation errors with multi-key aggs:
https://github.com/prestodb/presto/pull/16175/files#r657520172

If we are tracking partial history in HBO, we don't need to be that conservative and can allow more cases

Reply:
Thanks for clarification. I guess we will still keep this behaviour for CBO and only extend for HBO right?